### PR TITLE
Full refresh

### PR DESCRIPTION
**Pull request by mareklibra**
User can invoke 'Refresh' to fully reload data

### CODE REVIEW

File: src/reducers/vms.js [hunk 1]
```diff
@@ -27,6 +27,8 @@ function removeMissingVms ({ state, payload: { vmIdsToPreserve } }) {
 }
 
 function updateVmDisk ({ state, payload: { vmId, disk } }) {
+  // TODO: If disk management is implemented then maintain disks separately
+  // and manage references (like for icons). Recently it makes no sense
```
Comment (matobet) [inline src/reducers/vms.js]: s/Recently/For now :-)

File: src/reducers/vms.js [hunk 2]
```diff
@@ -35,6 +37,15 @@ function updateVmDisk ({ state, payload: { vmId, disk } }) {
   return state
 }
 
+function clearVmDisks ({ state, payload: { vmId } }) {
```
Comment (matobet) [inline src/reducers/vms.js]: just a question: why do we need to clear a VM's disks?

File: src/reducers/vms.js [hunk 3]
```diff
@@ -15,6 +15,17 @@ function removeVms ({ state, payload: { vmIds } }) {
   return mutable.asImmutable()
 }
 
+function removeMissingVms ({ state, payload: { vmIdsToPreserve } }) {
+  const vmIdsToBeDeleted = state.get('vms')
+    .keySeq()
+    .filter(vmId => !vmIdsToPreserve.some(toPreserveId => toPreserveId === vmId))
+    .map(vmId => vmId)
```
Comment (matobet) [inline src/reducers/vms.js]: the last mapping appears to be a redundant (identity) operation

**Review by matobet — CHANGES_REQUESTED** [review]

File: src/reducers/vms.js (hunk 3 above)
Comment (mareklibra) [reply src/reducers/vms.js]: Right, left over ...

**Comment by mareklibra** [comment]
Done

File: src/reducers/vms.js [hunk 4]
```diff
@@ -15,9 +15,19 @@ function removeVms ({ state, payload: { vmIds } }) {
   return mutable.asImmutable()
 }
 
-function updateVmDisk ({ state, payload: { vmId, disk } }) {
+function removeMissingVms ({ state, payload: { vmIdsToPreserve } }) {
+  const vmIdsToBeDeleted = state.get('vms')
+    .keySeq()
+    .filter(vmId => !vmIdsToPreserve.some(toPreserveId => toPreserveId === vmId))
```
Comment (matobet) [inline src/reducers/vms.js]: please consider using a `Set` instead of doing nested linear lookups to check for presence

**Review by matobet — APPROVED** [review]

**Pull request merged**